lib/cpp/src/thrift/server/TNonblockingServer.cpp

```diff
@@ -547,12 +553,18 @@ void TNonblockingServer::TConnection::transition() {
   case APP_READ_REQUEST:
     // We are done reading the request, package the read buffer into transport
     // and get back some data from the dispatch function
-    inputTransport_->resetBuffer(readBuffer_, readBufferPos_);
-    outputTransport_->resetBuffer();
-    // Prepend four bytes of blank space to the buffer so we can
-    // write the frame size there later.
-    outputTransport_->getWritePtr(4);
-    outputTransport_->wroteBytes(4);
+    if (!server_->getOutputProtocolFactory()) {
```

Comment:
If I understand this correctly, then this is basically asking "are we in the THeader case".  I think that this information should be stored in a well named member enum, and then we can query that.  The current query doesn't really express intent to me.
Alternatively, comment the reason for the difference.